x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformFailureHandler.java

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.transform.transforms;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.common.breaker.CircuitBreakingException;
+import org.elasticsearch.script.ScriptException;
+import org.elasticsearch.xpack.core.transform.TransformMessages;
+import org.elasticsearch.xpack.core.transform.transforms.SettingsConfig;
+import org.elasticsearch.xpack.core.transform.utils.ExceptionsHelper;
+import org.elasticsearch.xpack.transform.notifications.TransformAuditor;
+import org.elasticsearch.xpack.transform.utils.ExceptionRootCauseFinder;
+
+import java.util.Optional;
+
+import static org.elasticsearch.core.Strings.format;
+
+/**
+ * Handles all failures a transform can run into when searching, indexing as well as internal
+ * state handling.
+ *
+ * TODO:
+ *
+ *  - the settings have to be passed as parameter - because they can change at runtime - longer term the necessary
+ *    parts should be read from the context object instead
+ */
+class TransformFailureHandler {
+    private static final Logger logger = LogManager.getLogger(TransformFailureHandler.class);
+    public static final int LOG_FAILURE_EVERY = 10;
+    private final TransformAuditor auditor;
+    private final String transformId;
+    private final TransformContext context;
+
+    // protected for unit tests
```

Comment:
This comment implies that it's _only_ the unit tests that construct this class directly, when in fact another production class does too.

The constructor is package private and not commenting on that at all causes least confusion in my opinion.

Reply:
good spot. I think I did not comment the constructor, but some member which I deleted later. Re-formatting deleted the blankline and somehow this ended up here. 

It makes no sense, of course I use this.